include/picolibrary/wiznet/w5500.h

```diff
@@ -486,6 +486,1219 @@ class Communication_Controller : public Device {
+        static constexpr auto UCASTB = std::uint_fast8_t{ P + Size::P }; ///< UCASTB.
+        static constexpr auto ND = std::uint_fast8_t{ UCASTB + Size::UCASTB }; ///< ND.
+        static constexpr auto BCASTB = std::uint_fast8_t{ ND + Size::ND }; ///< BCASTB.
+        static constexpr auto MULTI = std::uint_fast8_t{ BCASTB + Size::BCASTB }; ///< MULTI.
+
+        static constexpr auto MIP6B = UCASTB; ///< MIP6B.
+
+        static constexpr auto MC  = ND; ///< MC.
+        static constexpr auto MMB = ND; ///< MMB.
+
+        static constexpr auto MFEN = MULTI; ///< MFEN;
+    };
+
+    /**
+     * \brief Field bit masks.
+     */
+    struct Mask {
+        static constexpr auto P      = mask<Type>( Size::P, Bit::P );           ///< P.
+        static constexpr auto UCASTB = mask<Type>( Size::UCASTB, Bit::UCASTB ); ///< UCASTB.
+        static constexpr auto ND     = mask<Type>( Size::ND, Bit::ND );         ///< ND.
+        static constexpr auto BCASTB = mask<Type>( Size::BCASTB, Bit::BCASTB ); ///< BCASTB.
+        static constexpr auto MULTI = mask<Type>( Size::MULTI, Bit::MULTI ); ///< MULTI.
+
+        static constexpr auto MIP6B = UCASTB; ///< MIP6B.
+
+        static constexpr auto MC  = ND; ///< MC.
+        static constexpr auto MMB = ND; ///< MMB.
+
+        static constexpr auto MFEN = MULTI; ///< MFEN;
+    };
+
+    /**
+     * \brief P.
+     */
+    enum P : Type {
+        P_CLOSED = 0b0000 << Bit::P, ///< Closed.
+        P_TCP    = 0b0001 << Bit::P, ///< TCP.
+        P_UDP    = 0b0010 << Bit::P, ///< UDP.
+        P_MACRAW = 0b0100 << Bit::P, ///< MACRAW.
+    };
+
+    /**
+     * \brief MC.
+     */
+    enum MC : Type {
+        MC_IGMP_VERSION_2 = 0 << Bit::MC, ///< IGMP version 2.
+        MC_IGMP_VERSION_1 = 1 << Bit::MC, ///< IGMP version 1.
+    };
+};
+
+/**
+ * \brief Socket N Command Register (SN_CR) register.
+ */
+struct SN_CR {
+    /**
+     * \brief Register type.
+     */
+    using Type = std::uint8_t;
+
+    /**
+     * \brief Register offset.
+     */
+    static constexpr auto OFFSET = Memory_Offset{ 0x0001 };
+
+    /**
+     * \brief Command.
+     */
+    enum COMMAND : Type {
+        COMMAND_OPEN      = 0x01, ///< Open.
+        COMMADN_LISTEN    = 0x02, ///< Listen.
+        COMMAND_CONNECT   = 0x04, ///< Connect.
+        COMMAND_DISCON    = 0x08, ///< Disconnect.
+        COMMAND_CLOSE     = 0x10, ///< Close.
+        COMMAND_SEND      = 0x20, ///< Send.
+        COMMAND_SEND_MAC  = 0x21, ///< Send UDP datagram using a manually configured destination MAC address instead of one obtained by ARP.
+        COMMAND_SEND_KEEP = 0x22, ///< Send TCP keepalive packet.
+        COMMAND_RECV      = 0x40, ///< Receive.
+    };
+};
+
+/**
+ * \brief Socket N Interrupt Register (SN_IR) register.
+ *
+ * This register has the following fields:
+ * - Peer Connected (CON)
+ * - Peer Disconnected (DISCON)
+ * - Data Received (RECV)
+ * - Timeout (TIMEOUT)
+ * - Send Complete (SEND_OK)
```

Comment:
Change `SEND_OK` to `SENDOK` (SN_IR an SN_IMR field names are inconsistent, `SENDOK` is more consistent with other field names).